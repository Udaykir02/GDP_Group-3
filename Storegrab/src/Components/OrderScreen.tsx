import React from 'react';
import { View, Text, StyleSheet, ScrollView, Image } from 'react-native';

interface Size {
    h: number;
    l: number;
    w: number;
}

interface Item {
    skuId: string;
    item: string;
    price: number;
    qty: number;
    size: Size;
    features: string;
    categories: string[];
    image: string;
    description: string;
    brand: string;
}

interface Address {
    street1: string;
    street2: string;
    city: string;
    state: string;
    country: string;
    zip: string;
}

interface Shipping {
    address: Address;
    origin: Address;
    carrier: string;
    tracking: string;
}

interface Order {
    orderId: string;
    userId: string;
    paymentId: string;
    vendorId: string;
    paymentStatus: string;
    paymentMethod: string;
    status: string;
    currency: string;
    totalCost: number;
    items: Item[];
    shipping: Shipping;
    orderTime: Date;
    endTime?: Date;
}

interface Props {
    order: Order;
}

const OrderScreen: React.FC<Props> = ({ navigation, route }: any) => {
    const { order }: Props = route.params;
    return (
        <ScrollView style={styles.container}>
            <Text style={styles.subHeader}>Order ID: {order.orderId}</Text>
            <Text>Status: {order.status}</Text>
            <Text>Payment Status: {order.paymentStatus}</Text>
            <Text>Total Cost: {order.currency} {order.totalCost}</Text>
            <Text>Order Time: {new Date(order.orderTime).toDateString()}</Text>
            {order.endTime && <Text>End Time: {new Date(order.endTime).toLocaleString()}</Text>}

            <View style={styles.section}>
                <Text style={styles.sectionHeader}>Shipping Information</Text>
                <Text>Carrier: {order.shipping.carrier}</Text>
                <Text>Tracking: {order.shipping.tracking}</Text>
                <Text>Address: {order.shipping.address.street1}, {order.shipping.address.street2}, {order.shipping.address.city}, {order.shipping.address.state}, {order.shipping.address.country} - {order.shipping.address.zip}</Text>
            </View>

            <View style={styles.section}>
                <Text style={styles.sectionHeader}>Items</Text>
                {order.items.map((item, index) => (
                    <View key={item.skuId} style={styles.item}>
                        <Image source={{ uri: item.image }} style={styles.itemImage} />
                        <View style={styles.itemDetails}>
                            <Text style={styles.itemHeader}>{item.item}</Text>
                            <Text>Brand: {item.brand}</Text>
                            <Text>Price: {order.currency} {item.price.toFixed(2)}</Text>
                            <Text>Quantity: {item.qty}</Text>
                            <Text>Size: {item.size.h}x{item.size.l}x{item.size.w}</Text>
                            <Text>Features: {item.features}</Text>
                            <Text>Categories: {item.categories.join(', ')}</Text>
                            <Text>Description: {item.description}</Text>
                        </View>
                    </View>
                ))}
            </View>
        </ScrollView>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 16,
        backgroundColor: '#fff',
    },
    header: {
        fontSize: 24,
        fontWeight: 'bold',
        marginBottom: 8,
    },
    subHeader: {
        fontSize: 18,
        marginBottom: 8,
    },
    section: {
        marginTop: 16,
        paddingBottom: 16,
        borderBottomWidth: 1,
        borderBottomColor: '#ddd',
    },
    sectionHeader: {
        fontSize: 20,
        fontWeight: 'bold',
        marginBottom: 8,
    },
    item: {
        flexDirection: 'row',
        marginBottom: 16,
    },
    itemImage: {
        width: 100,
        height: 100,
        marginRight: 16,
    },
    itemDetails: {
        flex: 1,
    },
    itemHeader: {
        fontSize: 18,
        fontWeight: 'bold',
    },
});

export default OrderScreen;
