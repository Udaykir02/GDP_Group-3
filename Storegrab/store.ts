import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import rootReducers from './reducers'; // Assuming rootReducers is defined in a separate file
import rootSaga from './sagas/rootSaga'; // Assuming rootSaga is defined in a separate file

// Create saga middleware instance
const sagaMiddleware = createSagaMiddleware();

// Create the Redux store with middleware and reducers
const store = configureStore({
  reducer: rootReducers,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
});

// Run the root saga
sagaMiddleware.run(rootSaga);

export default store;
