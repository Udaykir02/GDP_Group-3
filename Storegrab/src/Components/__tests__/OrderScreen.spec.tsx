import { Provider } from 'react-redux'
import store from '../../../store'
import render, {
    RenderResult,
} from '@testing-library/react-native/build/render'
import VendorSubscriptionPage from '../Subscription'
import { NavigationContainer } from '@react-navigation/native'
import OrderScreen from '../OrderScreen'
const mockOrder = {
    orderId: "ORD123456",
    userId: "USR78910",
    paymentId: "PAY09876",
    vendorId: "VND54321",
    paymentStatus: "Paid",
    paymentMethod: "Credit Card",
    status: "Shipped",
    currency: "USD",
    totalCost: 39.98,
    items: [
        {
            skuId: "12345",
            item: "Widget A",
            price: 19.99,
            qty: 2,
            size: {
                h: 10,
                l: 5,
                w: 3
            },
            features: "Durable, lightweight, water-resistant",
            categories: ["tools", "gadgets"],
            image: "http://example.com/images/widgetA.png",
            description: "A high-quality widget useful for various tasks.",
            brand: "WidgetCo"
        },
        {
            skuId: "67890",
            item: "Gadget B",
            price: 9.99,
            qty: 1,
            size: {
                h: 10,
                l: 5,
                w: 3
            },
            features: "Compact, multi-purpose",
            categories: ["gadgets"],
            image: "http://example.com/images/gadgetB.png",
            description: "A versatile gadget with multiple uses.",
            brand: "GadgetInc"
        }
    ],
    shipping: {
        address: {
            street1: "123 Main St",
            street2: "Apt 4B",
            city: "Springfield",
            state: "IL",
            country: "USA",
            zip: "62701"
        },
        origin: {
            street1: "789 Warehouse Rd",
            street2: "",
            city: "Shelbyville",
            state: "IL",
            country: "USA",
            zip: "62702"
        },
        carrier: "UPS",
        tracking: "1Z9999999999999999"
    },
    orderTime: new Date("2024-05-19T12:34:56.000Z"),
    endTime: new Date("2024-05-26T12:34:56.000Z")
}

jest.mock('@react-navigation/native', () => {
    return {
      ...jest.requireActual('@react-navigation/native'),
      useRoute: () => ({
        params: {
          order: mockOrder,
        },
      }),
    };
  });

const renderStartupContainerPage = () => {
    return render(
        <Provider store={store}>
            <NavigationContainer>
                <OrderScreen route={{params: { order: mockOrder}}} />
            </NavigationContainer>
        </Provider>,
    )
}
let page: RenderResult
describe('VendorSubscriptionPage', () => {
    beforeEach(() => {
        jest.useFakeTimers();
        page = renderStartupContainerPage()
    })
    afterEach(() => {
        jest.runOnlyPendingTimers();
        jest.useRealTimers();
    });
    it('should render correctly', function () {
        expect(page.toJSON()).toMatchSnapshot()
    })

});