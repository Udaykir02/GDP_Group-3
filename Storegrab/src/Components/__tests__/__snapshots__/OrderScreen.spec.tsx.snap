// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorSubscriptionPage should render correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 18,
          "marginBottom": 8,
        }
      }
    >
      Order ID: 
      ORD123456
    </Text>
    <Text>
      Status: 
      Shipped
    </Text>
    <Text>
      Payment Status: 
      Paid
    </Text>
    <Text>
      Total Cost: 
      USD
       
      39.98
    </Text>
    <Text>
      Order Time: 
      Sun May 19 2024
    </Text>
    <Text>
      End Time: 
      26/05/2024, 07:34:56
    </Text>
    <View
      style={
        {
          "borderBottomColor": "#ddd",
          "borderBottomWidth": 1,
          "marginTop": 16,
          "paddingBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 8,
          }
        }
      >
        Shipping Information
      </Text>
      <Text>
        Carrier: 
        UPS
      </Text>
      <Text>
        Tracking: 
        1Z9999999999999999
      </Text>
      <Text>
        Address: 
        123 Main St
        , 
        Apt 4B
        , 
        Springfield
        , 
        IL
        , 
        USA
         - 
        62701
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#ddd",
          "borderBottomWidth": 1,
          "marginTop": 16,
          "paddingBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 8,
          }
        }
      >
        Items
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 16,
          }
        }
      >
        <Image
          source={
            {
              "uri": "http://example.com/images/widgetA.png",
            }
          }
          style={
            {
              "height": 100,
              "marginRight": 16,
              "width": 100,
            }
          }
        />
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Widget A
          </Text>
          <Text>
            Brand: 
            WidgetCo
          </Text>
          <Text>
            Price: 
            USD
             
            19.99
          </Text>
          <Text>
            Quantity: 
            2
          </Text>
          <Text>
            Size: 
            10
            x
            5
            x
            3
          </Text>
          <Text>
            Features: 
            Durable, lightweight, water-resistant
          </Text>
          <Text>
            Categories: 
            tools, gadgets
          </Text>
          <Text>
            Description: 
            A high-quality widget useful for various tasks.
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 16,
          }
        }
      >
        <Image
          source={
            {
              "uri": "http://example.com/images/gadgetB.png",
            }
          }
          style={
            {
              "height": 100,
              "marginRight": 16,
              "width": 100,
            }
          }
        />
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Gadget B
          </Text>
          <Text>
            Brand: 
            GadgetInc
          </Text>
          <Text>
            Price: 
            USD
             
            9.99
          </Text>
          <Text>
            Quantity: 
            1
          </Text>
          <Text>
            Size: 
            10
            x
            5
            x
            3
          </Text>
          <Text>
            Features: 
            Compact, multi-purpose
          </Text>
          <Text>
            Categories: 
            gadgets
          </Text>
          <Text>
            Description: 
            A versatile gadget with multiple uses.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
